compiler/src/iree/compiler/GlobalOptimization/GeneralizeLinalgNamedOps.cpp

```diff
@@ -31,32 +31,37 @@ struct GeneralizeLinalgNamedOpsPass
 };
 } // namespace
 
-/// Returns true of `linalgOp` is a Conv2DNchwFchwOp or Conv2DNhwcHwcfOp with
-/// all strides equal to 1 and with a kernel height and width of 1
+/// Returns true if `linalgOp` is a convolution op with all strides equal to 1
+/// and with unit kernel dimensions.
 static bool isConvFoldableToContraction(linalg::LinalgOp linalgOp) {
-  auto NCHWOp = dyn_cast<linalg::Conv2DNchwFchwOp>(linalgOp.getOperation());
-  auto NHWCOp = dyn_cast<linalg::Conv2DNhwcHwcfOp>(linalgOp.getOperation());
-
-  if (!NCHWOp && !NHWCOp)
+  auto convDimsOrFailure = linalg::inferConvolutionDims(linalgOp);
+  if (failed(convDimsOrFailure)) {
     return false;
-
-  DenseIntElementsAttr strides =
-      NCHWOp ? NCHWOp.getStrides() : NHWCOp.getStrides();
-  if (!llvm::all_of(
-          strides, [](APInt element) { return element.getSExtValue() == 1; })) {
+  }
+  auto &convDims = *convDimsOrFailure;
+  if (!llvm::all_of(convDims.strides,
```

Comment:
nit : can use `if(!hasAllOneValues(..)` instead

Reply:
Actually ignore this, its a local function in the code I was looking at, so its fine the way it is.